core/services/relay/evm/codec_entry.go

```diff
@@ -0,0 +1,147 @@
+package evm
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/codec"
+
+	commontypes "github.com/smartcontractkit/chainlink-common/pkg/types"
+
+	"github.com/smartcontractkit/chainlink/v2/core/services/relay/evm/types"
+)
+
+type codecEntry struct {
+	Args           abi.Arguments
+	encodingPrefix []byte
+	checkedType    reflect.Type
+	nativeType     reflect.Type
+	mod            codec.Modifier
+}
+
+func (entry *codecEntry) Init() error {
+	if entry.checkedType != nil {
+		return nil
+	}
+
+	args := UnwrapArgs(entry.Args)
+	argLen := len(args)
+	native := make([]reflect.StructField, argLen)
+	checked := make([]reflect.StructField, argLen)
+
+	if len(args) == 1 && args[0].Name == "" {
+		nativeArg, checkedArg, err := getNativeAndCheckedTypes(&args[0].Type)
+		if err != nil {
+			return err
+		}
+		entry.nativeType = nativeArg
+		entry.checkedType = checkedArg
+		return nil
+	}
+
+	for i, arg := range args {
+		tmp := arg.Type
+		nativeArg, checkedArg, err := getNativeAndCheckedTypes(&tmp)
+		if err != nil {
+			return err
+		}
+		tag := reflect.StructTag(`json:"` + arg.Name + `"`)
+		name := strings.ToUpper(arg.Name[:1]) + arg.Name[1:]
```

Comment:
We can get a panic here when solidity returns multiple unnamed values

Reply:
eg.  this is a valid solidity function signature
` function name() external view returns (
        address,
        uint32,
        uint8,
        uint32
    )`